src/mds/MDS.cc

```diff
@@ -2333,6 +2333,19 @@ void MDS::handle_signal(int signum)
   mds_lock.Unlock();
 }
 
+void MDS::damaged()
+{
+  set_want_state(MDSMap::STATE_DAMAGED);
+  beacon.notify_health(this);  // Include latest status in our swan song
+  beacon.send();
```

Comment:
I think we might need to wait a bit here before quitting, or we could easily shut down before the message gets out of the tcp buffers. The OSD has some code to wait a short while on shutdown that we can probably crib from if you want.